### PR TITLE
docs(java): Add `@CdsName` to example

### PR DESCRIPTION
**Pull request by bugwelle**
A section in "Event Handlers" mentions `@CdsName`, but does not show any example.  This commit adds an example and fixes a few typos/grammar issues.

It also fixes the name `AddReviewContext`, which either changed in CAP Java or was wrong before, because for the bound action, `BooksAddReviewContext` is generated.

### CODE REVIEW

File: java/event-handlers/indicating-errors.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ throw new ServiceException(ErrorStatuses.CONFLICT, "Not enough stock available")
 throw new ServiceException(ErrorStatuses.BAD_REQUEST, "No book title specified", originalException);
 ```
 
-The OData V4 adapter turns all exceptions into an OData error response to indicate the error to the client.
+The OData adapters turn all exceptions into an OData error response to indicate the error to the client.
```
Comment (bugwelle) [inline java/event-handlers/indicating-errors.md:45]: I clarified this with @eugene-andreev : it's not just the OData v4 adapter.

File: java/event-handlers/indicating-errors.md [hunk 2]
```diff
@@ -273,7 +273,7 @@ The same applies to message targets that refer to an action or function input pa
 
 ``` java
 @Before
-public void validateReview(AddReviewContext context) {
+public void validateReview(BooksAddReviewContext context) {
```
Comment (bugwelle) [inline java/event-handlers/indicating-errors.md:276]: tested locally: `BooksAddReviewContext` is generated

File: java/event-handlers/request-contexts.md [hunk 3]
```diff
@@ -279,7 +279,8 @@ public class CustomUserInfoProvider implements UserInfoProvider {
             }
         }
         if (userInfo != null) {
-            userInfo.setName(userInfo.getName().toLowerCase()); // Normalize user name
+            // Normalize user name
+            userInfo.setName(userInfo.getName().toLowerCase(Locale.US));
```
Comment (bugwelle) [inline java/event-handlers/request-contexts.md:283]: `getName()` returns a string, right? If so, then the standard method `toLowerCase()` should get a locale

Comment (rjayasinghe) [reply java/event-handlers/request-contexts.md:283]: > `getName()` returns a string, right? If so, then the standard method `toLowerCase()` should get a locale

You're right. But I would take Locale.ENGLISH.

**Review by rjayasinghe — COMMENTED** [review]
small inline comment regarding locale for string operations.

File: java/event-handlers/request-contexts.md (hunk 3 above)
Comment (bugwelle) [reply java/event-handlers/request-contexts.md:283]: I used what I found in `cds-services` :D

But sure!

```suggestion
            userInfo.setName(userInfo.getName().toLowerCase(Locale.ENGLISH));
```

File: java/event-handlers/request-contexts.md [hunk 4]
```diff
@@ -162,7 +162,7 @@ public void afterHandler(EventContext context){
...
 
-The application offers an action for one of its CDS entities. Within the action, a communication happens with a remote CAP service using an internal technical user from the provider account. The corresponding `on` handler of the action needs to create a new Request Context by calling `requestContext()`. Using the `systemUserProvider()` method, the existing user information is removed and the tenant is automatically set to the provider tenant. This allows the application to perform an HTTP call to the remote CAP service, which is secured using the pseudo-role `internal-user`.
+The application offers an action for one of its CDS entities. Within the action, the application communicate with a remote CAP service using an internal technical user from the provider account. The corresponding `on` handler of the action needs to create a new Request Context by calling `requestContext()`. Using the `systemUserProvider()` method, the existing user information is removed and the tenant is automatically set to the provider tenant. This allows the application to perform an HTTP call to the remote CAP service, which is secured using the pseudo-role `internal-user`.
```
Comment (rjayasinghe) [inline java/event-handlers/request-contexts.md:165]: "the application communicates with .."

**Review by rjayasinghe — COMMENTED** [review]
only one small typo (comment inline)

File: java/event-handlers/request-contexts.md (hunk 4 above)
Comment (bugwelle) [reply java/event-handlers/request-contexts.md:165]:
```suggestion
The application offers an action for one of its CDS entities. Within the action, the application communicates with a remote CAP service using an internal technical user from the provider account. The corresponding `on` handler of the action needs to create a new Request Context by calling `requestContext()`. Using the `systemUserProvider()` method, the existing user information is removed and the tenant is automatically set to the provider tenant. This allows the application to perform an HTTP call to the remote CAP service, which is secured using the pseudo-role `internal-user`.
```

Comment (bugwelle) [reply java/event-handlers/request-contexts.md]: Oh, thanks! :)

**Review by smahati — APPROVED** [review]

**Pull request merged**